tests/DocBlock/TypeExpressionTest.php

```diff
@@ -417,6 +417,10 @@ public static function provideAllowsNullCases(): iterable
         yield ['bool', false];
 
         yield ['string', false];
+
+        yield ['?int', true];
+
+        yield ['bool|?\Closure(): void', true];
```

Comment:
From [RFC](https://wiki.php.net/rfc/union_types_v2#nullable_union_types):

> Union types and the `?T` nullable type notation cannot be mixed

It ends up with `Parse error: syntax error, unexpected token "?"`.

Reply:
Yeah, `#7000`!

Test fixed, makes sense to no add someting not natively supported.

Reply:
To my question about functional case, do you think we can add some functional/fixer test that will fail /wo this fix?

Reply:
`TypeExpression::allowsNull()` is used only in `AbstractPhpdocToTypeDeclarationFixer::getCommonTypeFromAnnotation()`, so probably test cases should be added to its child classes: `PhpdocToReturnTypeFixer`, `PhpdocToParamTypeFixer` and `PhpdocToPropertyTypeFixer`. Currently it fails, because in `AbstractPhpdocToTypeDeclarationFixer::getCommonTypeFromAnnotation()` there's check `Preg::match(self::CLASS_REGEX, $commonType)` which causes that `null` is returned and fix is skipped.

<img width="1840" alt="image" src="https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/assets/600668/4ef938ae-62fb-4477-bbf8-51a55ae2663b">

I rather don't see `?Foo` format usage in phpDoc (only union), but [it's valid too](https://phpstan.org/r/ed03e113-3624-4b3d-998d-225b48367d17), so it should be covered. Would like to fix `AbstractPhpdocToTypeDeclarationFixer` and provide test cases for those 3 mentioned fixers 🙂?

Reply:
Please push the tests to my PR branch.